webroot/src/network/licenseApi/data.api.ts

```diff
@@ -75,62 +78,45 @@ export class LicenseDataApi implements DataApiInterface {
         );
     }
 
-    /**
-     * Prep a URI query parameter string for GET requests.
-     * @param  {RequestParamsInterface} params The request query parameters config.
-     * @return {string}                        The URI query param string.
-     */
-    public prepRequestGetParams(params: RequestParamsInterface = {}): string {
-        let requestParams = '';
-
-        if (params.pageSize) {
-            requestParams += `?pageSize=${params.pageSize}`;
-        } else {
-            requestParams += `?pageSize=100`;
-        }
-
-        if (params.pageNumber) {
-            requestParams += `&pageNumber=${params.pageNumber}`;
-        }
-
-        if (params.sortBy) {
-            requestParams += `&sortBy=${params.sortBy}`;
-        }
-
-        if (params.sortDirection) {
-            requestParams += `&sortDir=${params.sortDirection}`;
-        }
-
-        return requestParams;
-    }
-
     /**
      * Prep a URI query parameter object for POST requests.
      * @param  {RequestParamsInterface} params The request query parameters config.
      * @return {object}                        The URI query param object.
      */
     public prepRequestPostParams(params: RequestParamsInterface = {}): string {
-        const requestParams: any = {};
-
-        if (params.pageSize || params.lastKey) {
-            requestParams.pagination = {};
+        const requestParams: any = { query: {}};
 
-            if (params.pageSize) {
-                requestParams.pagination.pageSize = params.pageSize;
-            }
-            if (params.lastKey) {
-                requestParams.pagination.lastKey = params.lastKey;
-            }
+        if (params.compact) {
+            requestParams.query.compact = params.compact;
         }
 
-        if (params.sortBy || params.sortDirection) {
-            requestParams.sorting = {};
+        if (params.jurisdiction) {
+            requestParams.query.jurisdiction = params.jurisdiction;
+        }
 
-            if (params.sortBy) {
-                requestParams.sorting.key = params.sortBy;
+        if (params.licenseeId) {
+            requestParams.query.providerId = params.licenseeId;
+        } else {
```

Comment:
> To confirm my understanding: Does the value of licenseeId determine if this is a detail or list view request?

Yep. I don't know if it's been settled yet if the server will use this same prop for search (upcoming); as in whether it could eventually accept partial licensee ids that could potentially return many records.

> Could we reduce nesting by one layer by just returning in the above if statement? If that goes against your style guide, ignore me.

We prefer not to have early `return`s.

Reply:
Got it. Thanks for the explanation. Good leaving as-is!